x-pack/plugins/uptime/public/components/monitor/ping_list/columns/ping_timestamp/step_image_popover.tsx

```diff
@@ -120,6 +120,12 @@ export const StepImagePopover: React.FC<StepImagePopoverProps> = ({
   isImagePopoverOpen,
 }) => {
   const [imageData, setImageData] = React.useState<string | undefined>(imgSrc || undefined);
+
+  // for legacy screenshots, when a new image arrives, we must overwrite it
+  if (imgSrc && imgSrc !== imageData) {
```

Comment:
Seems like this should be a useEffect rather, so that it only sets if `imgSrc`.

Reply:
++ good point, I noticed a regression caused by changes to the state default above, and I quickly added this code and it appeared to fix the issue. But it should be an effect.

Reply:
@dominiqueclarke see b30bfa87859a5affc2ad6b123506c747e1b90b9f.